### PR TITLE
SITES-1101: BEAN title handling that is SMRT.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Handles the new bean->title variable in the bean template more elegantly by creating new logic around the way blocks and beans create their. 

# Needed By (Date)
- ?

# Urgency
- Semi urgeant 

# Steps to Test

- Install stanford profile
- Enable bean module 1.11
- Create a new bean type and add a body field
- Create a new bean and place it into the left sidebar filling in each of the bean entity fields
- See no duplicate title
- Check out bean 1.13
- Clear cache and revisit page with the bean on it
- See no duplicate title and that the bean title is the BEAN's title not the blocks
- Edit the bean type's view modes
- Move the title field to hidden and save
- Clear all caches
- Review the page with the bean on it again and see that the title is now owned by the block and not the bean
- Edit the block title configuration and save
- Refresh page and see new block title and not the bean title.

*Optional:*
- Play around with setting and not setting the block title to `<none>`
- Play around with setting and not setting content in to the title field on beans
- Play around with other view modes

# Affected Projects or Products
- Everything?

# Associated Issues and/or People
- https://www.drupal.org/project/bean/issues/2996282 
- SITES-1101

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
@pookmish Please to review as well.

**Comment by pookmish** [comment]
This breaks if the user has css applied to the old markup. 
books no title
screenshot double

https://jaaker.people.stanford.edu/research

**Comment by sherakama** [comment]
🤔 So this needs to default to the block title first, then the bean title. Ooof. That is going to be a problem.

**Comment by jbickar** [comment]
@pookmish thank you. This is a big problem.

**Comment by jbickar** [comment]
Can we thus expedite https://github.com/SU-SWS/acsf-leland/pull/37 ?

**Comment by sherakama** [comment]
@pookmish @jbickar 

After going around in circles for a while and digging in to this issue on https://www.drupal.org/project/bean/issues/2996282

We are pretty much right back to where Mike had us to start with except that this is now in the Profile and not the theme. I'm up for weighing the pros vs cons to having this here vs in a bean template in all of our themes. As we are going to allow for custom themes that may or may not be based themed off of one of the provided themes it would be nice to have this in the profile. It doesn't appear to be borking anything by doing this here. On the flip side, it may be breaking something that is expecting the title property (not label it is different in beans) that I don't know about. Please state if you feel strongly about one way or the other or if no1curr.

TLDR; Drops title in hook_preprocess_entity. Functions just like Mike's template fix but in the profile.

**Comment by sherakama** [comment]
@pookmish, @jbickar  knock knock?

**Comment by pookmish** [comment]
Do we still need this since we reverted the bean module?

**Comment by sherakama** [comment]
Yes, as we want to get back up to the stable release. I would prefer not to be locked on 1.11 forever.

**Comment by jbickar** [comment]
Yeah, I agree with Shea, although I was hoping for contrib to get its act together and un-CF itself on this one.

Since this is not on fire, it has not drawn my attention. Eventually.

**Comment by sherakama** [comment]
unmbi2rmlla21

**Comment by sherakama** [comment]
Bean

**Comment by pookmish** [comment]
So this appears to work. I didnt follow all the steps as it didnt seem necessary. After updating bean, i got the duplicate title. then checked out this branch, cleared cache and removed the duplicate.

**Review by pookmish — APPROVED** [review]

**Comment by sherakama** [comment]
Ping @jbickar

**Comment by jbickar** [comment]
_LE SIGH_

**Comment by sherakama** [comment]
_**4 MONTHS**_, this PR is approaching 4 MONTHS.

**Comment by jbickar** [comment]
I have older ones. Do we want to make this a competition?

**Comment by sherakama** [comment]
This PR gets less love and attention than my first born will after my second child arrives.

**Comment by pookmish** [comment]
I gave it some love :D

**Comment by swberg** [comment]
This thread gets funnier every time I read it. 

_Takes a few notes for IPPs..._

**Comment by sherakama** [comment]
Status update: My wife has grown and had a small human exit her.

**Comment by sherakama** [comment]
POKE

**Comment by jbickar** [comment]
There's going to come a time when someone suggests that you focus on Drupal 8.

**Comment by sherakama** [comment]
00100dPORTRAIT_00100_BURST20190705075933046_COVER
Could this be the month that this PR gets merged wonders my 3-month-old child.

**Comment by jbickar** [comment]
I was hoping your Director would get you interested in Drupal 8 so you would forget about this. No such luck.

**Comment by sherakama** [comment]
raechicken

My 2nd born went as a chicken for Halloween, which leads me to wonder. What came first? The chicken, the egg, or this pull request?

**Comment by pookmish** [comment]
definitely the PR came first

**Comment by jbickar** [comment]
I don't see any urgency for this until there is a required security update for BEAN.

**Comment by sherakama** [comment]
At which point you will be burdened by all of this technical debt and a short timeline to resolve.

**Comment by pookmish** [comment]
oh snap

**Comment by jbickar** [comment]
That's a mighty big "if".

live-dangerously

**Comment by jbickar** [comment]
https://live.staticflickr.com/139/319073031_a5043c89be.jpg

**Comment by jbickar** [comment]
BE5B38FD-8FD2-4E2A-A4CB-E25C6BC13EDE

**Comment by jbickar** [comment]
This thread probably should be required reading for new developers.

**Comment by sherakama** [comment]
This pull request is in quarantine. Must have tested positive for COVID-19. No one wants to touch it.

**Comment by pookmish** [comment]
rofl

**Comment by sherakama** [comment]
I know what I am going to be for Halloween this year... An SA for BEAN 7.x-1.x... Boy would that be scary?

**Comment by jbickar** [comment]
If you want to get into PsyOps with me, all I can say is, "Go ahead: poke the bear."

**Comment by sherakama** [comment]
Happy 2nd Birthday Pull Request #117

**Comment by jbickar** [comment]
The longer this goes on, the more it proves me right.

**Comment by sherakama** [comment]
Happy 3rd birthday PR #117

**Comment by jbickar** [comment]
> The longer this goes on, the more it proves me right.

**Comment by sherakama** [comment]
_FINE_

**Comment by pookmish** [comment]
Shea's reaction:
fine

**Comment by jbickar** [comment]
I would just like to point out that I won, here. You guys got bored and wandered off.

**Comment by jbickar** [comment]
I have wrapped every single comment in this thread in aluminum foil. And for good measure, I used double-sided tape to ensure that the aluminum foil really stuck.

Boy, is that going to be a pain to clean up. I just can't imagine.
